### PR TITLE
ci: cron ci must use the logical AND condition

### PR DESCRIPTION
**Pull request by Chever-John**
### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [x] CI/CD or Tests

### What this PR does / why we need it:

Maybe there is an error in the ci logic.
link #1849

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.sheincorp.cn/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: .github/workflows/e2e-test-ci-v2-cron-dev.yml [hunk 1]
```diff
@@ -26,19 +26,14 @@ on:
...
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}-dev
+  group: ${{ github.workflow }}-dev
   cancel-in-progress: true
 
 jobs:
   changes:
-    if: ${{ (!github.event.pull_request.draft || github.event.schedule == '0 0 * * *') }}
+    if: ${{ github.event.schedule == '0 0 * * *') }}
```
Comment (tao12345666333) [inline .github/workflows/e2e-test-ci-v2-cron-dev.yml:36]: we can just remove this condition.

**Review by tao12345666333 — COMMENTED** [review]
Thanks

File: .github/workflows/e2e-test-ci-v2-cron-dev.yml (hunk 1 above)
Comment (Chever-John) [reply .github/workflows/e2e-test-ci-v2-cron-dev.yml:36]: Can I refer to any file?
Is this line used to run ci at regular intervals?

Comment (tao12345666333) [reply .github/workflows/e2e-test-ci-v2-cron-dev.yml:36]: As the trigger condition has already been specified through the "on" clause at the beginning of the file, this judgment is unnecessary.

Comment (Chever-John) [reply .github/workflows/e2e-test-ci-v2-cron-dev.yml]: Yeah, There is an update!

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
CI errors are unrelated to modifications. We can skip it.

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**